### PR TITLE
On 1389 staff app search performance

### PR DESCRIPTION
**Pull request by alandburns**

### CODE REVIEW

**Comment by alandburns** [comment]
Nasty query fix to get either Ilrs records or certificates in the Staff Search.

File: src/SFA.DAS.AssessorService.Application/Handlers/Staff/StaffSearchHandler.cs [hunk 1]
```diff
@@ -96,9 +96,9 @@ private async Task<StaffReposSearchResult> Search(StaffSearchRequest request)
                 // Search string is a long of 10 length so must be a uln.
                 var sr = new StaffReposSearchResult
```
Comment (martin-hornsby) [inline src/SFA.DAS.AssessorService.Application/Handlers/Staff/StaffSearchHandler.cs]: This code works, albeit is a bit smelly. Just wondering if this looks cleaner:

var searchResults = (await _staffIlrRepository.SearchForLearnerByUln(request)).ToList();
return new StaffReposSearchResult
                {
                    PageOfResults = searchResults,  
                    TotalCount = searchResults .Count
                 };

Comment (martin-hornsby) [inline src/SFA.DAS.AssessorService.Application/Handlers/Staff/StaffSearchHandler.cs]: This code works, albeit is a bit smelly. Just wondering if this looks cleaner:

var results = await _staffIlrRepository.SearchForLearnerByUln(request);

return new StaffReposSearchResult
{
	PageOfResults = results,
	TotalCount = results.Count()
};

File: src/SFA.DAS.AssessorService.Application/Handlers/Staff/StaffSearchHandler.cs [hunk 2]
```diff
@@ -107,9 +107,9 @@ private async Task<StaffReposSearchResult> Search(StaffSearchRequest request)
...
```
Comment (martin-hornsby) [inline src/SFA.DAS.AssessorService.Application/Handlers/Staff/StaffSearchHandler.cs]: This code works, albeit is a bit smelly. Just wondering if this looks cleaner:

var results = await _staffIlrRepository.SearchForLearnerByCertificateReference(request.SearchQuery);

return new StaffReposSearchResult
{
	DisplayEpao = true,
        PageOfResults = results,
	TotalCount = results.Count()
};

**Pull request merged**